sdk/Sdk.Generators/Constants.cs

```diff
@@ -5,15 +5,20 @@ namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
 {
     internal static class Constants
     {
+        internal static class ExecutionModel
+        {
+            internal const string Isolated = "isolated";
+        }
+
         internal static class Languages
         {
             internal const string DotnetIsolated = "dotnet-isolated";
         }
 
         internal static class BuildProperties
         {
+            internal const string FunctionsExecutionModel = "build_property.FunctionsExecutionModel";
```

Comment:
I don't think this will work in the end, as `FunctionsExecutionModel` is set in: https://github.com/Azure/azure-functions-dotnet-worker/blob/5ff83ce0fdf244536cea093686540b424dbfc1ea/sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.props#L14

and due to the transitive reference, this will also be part of the xunit project, so the condition will pass and generators still run.

We need to find something that is different between the xunit project (rather a non-functions project) and the functions project. But that is hard because bringing in the functions SDK at all _is_ our signal that this is a function project! We can brainstorm something else to use here, but also need to ensure it doesn't false disable the source generator unintetnionally.

May we can check if the project is a library vs executable?

Reply:
@jviau  Are you sure about this? My understanding is that the MSBuild properties are not propagated through project references, unless one explicitly do that via `<Import Project="path\to\your\propsfile.props" />` in the second project (in our case - the Test project)

My local validation also confirms this understanding. Are you still able to see the source generation happening inside the test projects when the artifacts from this PR is used? It is very much possible I missed something in my testing.

Reply:
<img width="348" alt="image" src="https://github.com/Azure/azure-functions-dotnet-worker/assets/144469/7a5bb323-15b3-402b-bba3-544558c0a529">